### PR TITLE
Runtime env

### PR DESCRIPTION
**Pull request by ramarag**
@jkotas @weshaggard  PTAL
@kouvel @rahku  

disregard the tests, will add them properly in corefx

### CODE REVIEW

File: src/dlls/mscoree/mscorwks_unixexports.src [hunk 1]
```diff
@@ -98,6 +98,7 @@ UnlockFile
 UnmapViewOfFile
 VirtualAlloc
 VirtualFree
+GlobalMemoryStatusEx
```
Comment (ramarag) [inline src/dlls/mscoree/mscorwks_unixexports.src]: i missed when i made the change https://github.com/dotnet/coreclr/commit/d973d04a9295b06723ab1e5d4fcd81525417800e

File: src/vm/ecalllist.h [hunk 2]
```diff
@@ -322,7 +322,7 @@ FCFuncStart(gRuntimeEnvironmentFuncs)
 #ifndef FEATURE_CORECLR
     QCFuncElement("_GetSystemVersion", SystemNative::_GetSystemVersion)
 #endif
-#if defined(FEATURE_CLASSIC_COMINTEROP) && !defined(FEATURE_CORECLR)
+#if defined(FEATURE_CLASSIC_COMINTEROP)
     QCFuncElement("GetRuntimeInterfaceImpl", SystemNative::GetRuntimeInterfaceImpl)
```
Comment (jkotas) [inline src/vm/ecalllist.h]: This should just return NULL for CoreCLR.

Comment (ramarag) [inline src/vm/ecalllist.h]: GetRuntimeInterfaceImpl throws for coreclr COMexception with E_NOINTERFACE  that translates to Invalidcastexception

File: src/dlls/mscoree/mscorwks_unixexports.src (hunk 1 above)
Comment (janvorli) [inline src/dlls/mscoree/mscorwks_unixexports.src]: Ah, ok

**Comment by jkotas** [comment]
There is nothing interesting in this class. It should be implemented in corefx. Let me know if you see anything problematic ... here are the ones that come to my mind:

```
GetRuntimeDirectory: Path.GetDirectoryName(new Uri(typeof(object).Assembly.CodeBase).AbsolutePath) + Path.DirectorySeparatorChar
GetRuntimeInterfaceImpl -> just throw the exception
SystemConfigurationFile -> throw platform not supported exception - does not make sense on CoreCLR
```

**Pull request closed without merging**